src/resources/items/ItemList.tsx

```diff
@@ -221,33 +223,54 @@ export default function ItemList(props?: ItemListType): React.ReactElement {
       {...props}>
       <ResetDateFilter source='createdAt' />
       {/* <ResetDateRangeFilter source='date_range' /> */}
-      <DatagridConfigurable
-        rowClick='show'
-        bulkActionButtons={<BulkActions />}
-        omit={omitColumns}>
-        <TextField source='item_number' label='Reference' />
-        <TextField source='id' />
-        <TextField source='createdAt' label='Created' />
-        <TextField source='mediaType' label='Media type' />
-        <SourceField
-          link='show'
-          source='loanedTo'
-          reference={constants.R_USERS}
-          label='Loaned to'
-        />
-        <DateField showTime source='start' />
-        <DateField showTime source='end' />
-        <SourceField source='vaultLocation' reference='vaultLocation' />
-        <SourceField source='protectiveMarking' reference='protectiveMarking' />
-        <SourceField
-          link='show'
-          source='batchId'
-          reference={constants.R_BATCHES}
-          sourceField='batchNumber'
-        />
-        <TextField source='remarks' />
-        <TextField source='musterRemarks' />
-      </DatagridConfigurable>
+      <DataList />
     </List>
   )
 }
+
+const DataList = (): React.ReactElement => {
+  const { onSelect, selectedIds } = useListContext()
+  const redirect = useRedirect()
+
+  const handleClick = (id: number): void => {
+    if (selectedIds.includes(id) === false) onSelect([...selectedIds, id])
+    else onSelect(selectedIds.filter((selectedId) => selectedId !== id))
+  }
+
+  const handleDoubleClick = (id: number): void => {
+    const path = `/${constants.R_ITEMS}/${id}/show`
+    redirect(path)
+  }
+
+  const handleRowClick = useDoubleClick(handleClick, handleDoubleClick)
```

Comment:
@TahaKhanAbdalli  - is it an option to pass in the (optional) double-click path destination to `useDoubleClick()`. That will reduce a little duplication.

Reply:
I was trying to keep them `decoupled` so that `useDoubleClick` hook doesn't know what to do on `single click` or `double click`. But if you say so I can handle pass the path to `useDoubleClick` hook.

Reply:
Yes, there isn't a clear single route through refactoring.

Could we see what it looks like with these capabilities pushed into `useDoubleClick` please?  The value in getting this implementation as "thin" as possible is that it makes it quicker/easier to apply it to other lists, too.

Reply:
Yes, there isn't a clear single route through refactoring.

Could we see what it looks like with these capabilities pushed into `useDoubleClick` please?  The value in getting this implementation as "thin" as possible is that it makes it quicker/easier to apply it to other lists, too.